core/tests/test_ingest.py

```
@@ -264,3 +269,76 @@ def test_per_glyph_mask_matches_full_page_mask(method):
+
+# ---------------------------------------------------------------------------
+# Memory-bounded threshold helpers
+# ---------------------------------------------------------------------------
+#
+# These exist only to keep peak memory off the page-area curve (a full-page
+# Sauvola on a 32 MP folio peaks over 1.5 GB and OOM-kills IC's 1 GiB
+# container). The whole point is that they change *nothing* about the output,
+# so that equality is what these tests pin.
+
+
+def _naive_sauvola(page):
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `_naive_sauvola`

(ANN202)